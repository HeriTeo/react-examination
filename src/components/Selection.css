:local(.selectOption) {
  	float: right;
  	width: 100%;
  	margin: 5px 0px 5px 0px;
}