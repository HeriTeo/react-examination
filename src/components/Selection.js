import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Selection.css';

class Selection extends Component {
  render() {
    const { onSelectData } = this.props;
    return (
      <select className={styles.selectOption} onChange={e => onSelectData(e.target.value)}>
        <option>ALL</option>
        <option>SF</option>
        <option>SG</option>
        <option>PF</option>
        <option>PG</option>
      </select>
    );
  }
}

Selection.propTypes = {
  onSelectData: PropTypes.func.isRequired,
};

Selection.defaultProps = {
  onSelectData: () => {},
};

export default Selection;
