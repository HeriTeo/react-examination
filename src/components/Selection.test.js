import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import Selection from './Selection';

configure({ adapter: new Adapter() });

describe('Selection', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<Selection />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
