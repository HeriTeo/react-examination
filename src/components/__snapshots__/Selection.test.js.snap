// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection renders without crashing 1`] = `
<select
  onChange={[Function]}
>
  <option>
    ALL
  </option>
  <option>
    SF
  </option>
  <option>
    SG
  </option>
  <option>
    PF
  </option>
  <option>
    PG
  </option>
</select>
`;
